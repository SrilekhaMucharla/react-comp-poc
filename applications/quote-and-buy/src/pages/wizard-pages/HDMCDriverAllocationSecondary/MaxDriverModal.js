import React from 'react';
import PropTypes from 'prop-types';
import { AnalyticsHDModal as HDModal } from '../../../web-analytics';
import * as messages from './HDMCDriverAllocationSecondary.messages';

const MaxDriverModal = ({
    onCancel, onConfirm, onClose, show, pageMetadata, hideClose
}) => (
    <HDModal
        webAnalyticsView={{ ...pageMetadata, page_section: messages.reachedMax }}
        webAnalyticsEvent={{ event_action: messages.reachedMax }}
        id="maximum-driver-count-modal"
        customStyle="rev-button-order"
        show={show}
        headerText={messages.important}
        cancelLabel={messages.noGoBack}
        confirmLabel={messages.ok}
        onCancel={onCancel}
        onConfirm={onConfirm}
        onClose={onClose}
        hideClose={hideClose}
    >
        <span>{messages.reachedMax}</span>
    </HDModal>
);

MaxDriverModal.propTypes = {
    onCancel: PropTypes.func.isRequired,
    onConfirm: PropTypes.func.isRequired,
    onClose: PropTypes.func.isRequired,
    show: PropTypes.bool,
    pageMetadata: PropTypes.shape({
        page_name: PropTypes.string.isRequired,
        page_type: PropTypes.string.isRequired,
        sales_journey_type: PropTypes.string.isRequired
    }).isRequired,
    hideClose: PropTypes.bool,
};

MaxDriverModal.defaultProps = {
    show: false,
    hideClose: false
};

export default MaxDriverModal;
