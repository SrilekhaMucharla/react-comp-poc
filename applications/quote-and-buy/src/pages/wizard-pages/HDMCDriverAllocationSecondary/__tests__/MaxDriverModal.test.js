import React from 'react';
import { shallow } from 'enzyme';
import MaxDriverModal from '../MaxDriverModal';

const pageMetadata = { page_name: 'test', page_type: 'test', sales_journey_type: 'multi_car' };

describe('MaxDriverModal', () => {
    // given
    const onCancel = jest.fn();
    const onConfirm = jest.fn();
    const onClose = jest.fn();
    const headerText = 'You can only add up to five people as drivers for this car.';
    const id = 'maximum-driver-count-modal';

    const getWrapper = (props = {}) => shallow(
        <MaxDriverModal
            onCancel={onCancel}
            onConfirm={onConfirm}
            onClose={onClose}
            headerText={headerText}
            pageMetadata={pageMetadata}
            show
            {...props} />
    );

    it('should render correctly and match the snapshot', () => {
        // given
        const wrapper = getWrapper();
        // then
        expect(wrapper).toMatchSnapshot();
    });

    it('should display HDModal', () => {
        // given
        const wrapper = getWrapper();
        // then
        expect(wrapper.find(`#${id}`)).toHaveLength(1);
    });

    it('should contain driver name', () => {
        // given
        const wrapper = getWrapper();
        // then
        expect(wrapper.find('span').text()).toContain(headerText);
    });

    it('should contain "this driver"', () => {
        // given
        const noDriverString = 'You can only add up to five people as drivers for this car.';
        const wrapper = shallow(
            <MaxDriverModal
                onCancel={onCancel}
                onConfirm={onConfirm}
                onClose={onClose}
                headerText=""
                pageMetadata={pageMetadata}
                show />
        );
        // then
        expect(wrapper.find('span').text()).toContain(noDriverString);
    });

    it('should pass functions to HDModal', () => {
        // given
        const wrapper = getWrapper();
        const modalProps = wrapper.find(`#${id}`).props();
        // then
        expect(modalProps.onCancel).toBe(onCancel);
        expect(modalProps.onClose).toBe(onClose);
        expect(modalProps.onConfirm).toBe(onConfirm);
    });
});
