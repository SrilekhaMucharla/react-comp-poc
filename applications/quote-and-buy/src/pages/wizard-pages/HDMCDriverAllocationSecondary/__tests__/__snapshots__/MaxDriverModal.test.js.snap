// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaxDriverModal should render correctly and match the snapshot 1`] = `
<ForwardRef
  cancelLabel="Go back"
  confirmLabel="OK"
  customStyle="rev-button-order"
  headerText="Important"
  hideClose={false}
  id="maximum-driver-count-modal"
  onBeforeOpen={null}
  onCancel={[MockFunction]}
  onClose={[MockFunction]}
  onConfirm={[MockFunction]}
  show={true}
  webAnalyticsEvent={
    Object {
      "event_action": "You can only add up to five people as drivers for this car.",
    }
  }
  webAnalyticsView={
    Object {
      "page_name": "test",
      "page_section": "You can only add up to five people as drivers for this car.",
      "page_type": "test",
      "sales_journey_type": "multi_car",
    }
  }
>
  <span>
    You can only add up to five people as drivers for this car.
  </span>
</ForwardRef>
`;
